crowbar_framework/app/controllers/setups_controller.rb

```diff
@@ -22,7 +22,31 @@ def show
   end
 
   def stop
-    # TODO: place function call
+    # Stop services at the nodes
+    @service_object = CrowbarService.new logger
+
+    @service_object.shutdown_openstack_at_nodes
+
+    # Dump of the database could fail with the message that there's not enough space
+    # We need to show the error to user so he can manually prepare the space on node
+    # and than redo this step.
+    begin
+      @service_object.dump_database
+    rescue => e
+      flash[:alert] = e.message
+      # FIXME: redirect to previous step?
+    end
+
+    # FIXME: now user needs to be told where is the database dump located in case
+    # he/she wants to manually fetch it
+    redirect_to setup_path
+  end
+
+  def finalize_stop
```

Comment:
I would name it just `finalize` and the route in the routes.rb is missing. https://github.com/jsuchome/crowbar-core/blob/a444d3d6689496e188df103550f046b2b55e0a7b/crowbar_framework/config/routes.rb#L148